### ISSUE TITLE
resolve() on Windows breaking change

### ISSUE THREAD

**Issue by blayz3r** [scrooloose/nerdtree#998]
### Environment

* Operating System: 
* Vim version `:version`: 
* NERDTree version `git rev-parse --short HEAD`: 
* NERDTree settings applied in your vimrc, if any:
    ```vim
    ```

### Process

1. 

### Possible Fix

https://groups.google.com/forum/#!topic/vim_dev/_g-_Kk6BXbk

**Comment by mrdubya** [comment]
Here is my follow up from the vim dev email.

### Environment
- OS: Windows 10,
- VIM version: 8.1.1417 or later
- NERDTree version: 67fa9b3,
- NERDTree setting: none.

As per the linked email thread the problem is that NERDTree is showing all simple files and directories as symbolic links as shown in this screenshot:

![NERD_tree_1 - (H__vim_vim) - GVIM 04_06_2019 17_23_42](https://user-images.githubusercontent.com/3754732/58944695-4979f500-877a-11e9-9148-e551626fd52b.png)

The problem is with the test for a symbolic link in Path.readInfoFromDisk(). hardPath uses a unix directory separator while VIM's resolve() now returns a full Windows path for the given file name. This results in the symlink test passing due to the different in the last directory separator.

For me the simplest fix that resolves the issue seems to be the following patch but this may not be the NERDTree way.

```diff --git a/lib/nerdtree/path.vim b/lib/nerdtree/path.vim
index c3af195..c1128e3 100644
--- a/lib/nerdtree/path.vim
+++ b/lib/nerdtree/path.vim
@@ -644,7 +644,7 @@ function! s:Path.readInfoFromDisk(fullpath)
     let lastPathComponent = self.getLastPathComponent(0)
 
     "get the path to the new node with the parent dir fully resolved
-    let hardPath = s:Path.Resolve(self.strTrunk()) . '/' . lastPathComponent
+    let hardPath = s:Path.Resolve(self.strTrunk()) . s:Path.Slash() . lastPathComponent
 
     "if  the last part of the path is a symlink then flag it as such
     let self.isSymLink = (s:Path.Resolve(hardPath) != hardPath)

```

**Comment by mrdubya** [comment]
VIM's `resolve()` has been reverted to not return a full Windows path for simple files.  This was fixed in VIM 8.1.1473.

While the suggested fix is no longer needed it would at least make the code more resiliant to unexpected VIM behaviour on Windows in the future.
